express-notes-api/index.js

```diff
@@ -0,0 +1,112 @@
+const express = require('express');
+const app = express();
+const data = require('./data');
+const fs = require('fs');
+const unexpectedErrorMsg = {
+  error: 'An unexpected error occurred.'
+};
+const notPositiveIntMsg = {
+  error: 'id must be a positive integer'
+};
+const needContentMsg = {
+  error: 'content is a required field'
+};
+
+app.use(express.json());
+
+app.get('/api/notes', (req, res) => {
+  const notesList = [];
+  for (const key in data.notes) {
+    notesList.push(data.notes[key]);
+  }
+  res.json(notesList);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const idNotFoundMsg = {
+    error: `cannot find note with id ${req.params.id}`
+  };
+
+  if (req.params.id < 0) {
+    res.status(400).json(notPositiveIntMsg);
+  } else if (data.notes[req.params.id] === undefined) {
+    res.status(404).json(idNotFoundMsg);
+  } else if (req.params.id) {
+    for (const key in data.notes) {
+      if (key === req.params.id) {
+        res.status(200).json(data.notes[req.params.id]);
+      }
+    }
+  }
```

Comment:
Since you have the `req.params.id`, you can access `data.notes` directory without the loop.

```js
const note = data.notes[req.params.id];
```